addons/calendar/static/src/js/calendar_controller.js

```diff
@@ -24,6 +24,30 @@ odoo.define('calendar.CalendarController', function (require) {
                 }).open();
             });
         },
+        /**
+         * Save the state of the filter in the database
+         * and continue after the save.
+         * @override
+         * @param {OdooEvent} event 
+         */
+        async _onChangeFilter(event) {
+            const _super = this._super; // keep the context inside the arrow function below
+            if (event.data.value !== "all") {
+                const domain = [['user_id', '=', this.getSession().uid], ['partner_id', '=', parseInt(event.data.value)]];
+                const contact = await this._rpc({
+                    model: 'calendar.contacts',
+                    method: 'search',
+                    args: [domain],
+                });
+
+                await this._rpc({
+                    model: 'calendar.contacts',
+                    method: 'write',
+                    args: [contact, {partner_checked: event.data.active}],
+                });
+            }
+            _super.apply(this, arguments);
```

Comment:
Don't need to hold a reference to "_super" anymore as it's not called in another scope, right?

I'm wondering what "super" exactly does but it would be nice if we could make the process of updating the filter in DB completely asynchronous from the rest (right now it's synchronous with the "awaits"). Not sure if it's possible but it would be worth checking for increased "performance" and usability.